### PR TITLE
Fixes lavaland flora.

### LINKED ISSUE #44519: Lavaland flora chemicals not extractable via botany disks
Issue reported from  Round ID: 110576 (/tg/Station Event Hall: Sseth overflow event. [ENGLISH] [EU] [100% FREE LAG])

Reporting client version: 512.1448

The special chemicals from lavaland plants, like tinia luxor and vitrium froth, cannot be extracted from the plants in the DNA extractor with a disk, unlike every other plant. They seem to only have a placeholder nutriment gene.

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

Fixes #44519

Lavaland plants now get their reagents from the plant genes and they now have the fireproof gene.

Napalm now burns weeds, even if the plant is fireproof.

Also moves all the lavaland seeds into ash_flora.dm.

## Why It's Good For The Game

#28533 seems to have intended this behaviour and it's consistent, I guess.

## Changelog
:cl:
add: Lavaland plants now have the fireproof gene.
tweak: Napalm now always burns weeds, even if the plant is fireproof.
fix: Lavaland plants now get their reagents from their genes.
/:cl:

### CODE REVIEW

**Comment by habader** [comment]
thanks

**Comment by habader** [comment]
could you also tie wheat's flour production to reagent too

**Comment by FantasticFwoosh** [comment]
``adjustWeeds(-rand(5,9)) //At least give them a small reward if they bother.``

syntax typo [here?](https://github.com/tgstation/tgstation/pull/44558/files#diff-cab7dc22dcade3f0b094b78a1de09185) it isn't in line with the rest of the code.

> could you also tie wheat's flour production to reagent too

You grind wheat stalks which have reagent flour within them % to the potency of the plant. Values are visible and editable from the plant gene editor typically in hydro. :question:  Im confused how to further help you.

**Comment by Time-Green** [comment]
cactus juice

File: code/modules/mining/lavaland/ash_flora.dm [hunk 1]
```diff
@@ -160,41 +164,130 @@
...
+	species = "ember"
+	plantname = "Embershroom Mushrooms"
+	product = /obj/item/reagent_containers/food/snacks/grown/ash_flora/mushroom_stem
+	genes = list(/datum/plant_gene/trait/plant_type/fungal_metabolism, /datum/plant_gene/trait/glow, /datum/plant_gene/trait/fire_resistance)
+	growing_icon = 'icons/obj/hydroponics/growing_mushrooms.dmi'
+	reagents_add = list(/datum/reagent/consumable/tinlux = 0.04, /datum/reagent/consumable/nutriment/vitamin = 0.02, /datum/reagent/drug/space_drugs = 0.02)
+
+//CRAFTING
+
+/datum/crafting_recipe/mushroom_bowl
```
Comment (Tlaltecuhtli) [inline code/modules/mining/lavaland/ash_flora.dm]: why crafting here there is a crafting recipe file

**Pull request merged**